packages/netlify-cms-widget-relation/src/RelationControl.js

```diff
@@ -111,22 +115,18 @@ export default class RelationControl extends React.Component {
     const initialSearchValues = value && (this.isMultiple() ? getSelectedOptions(value) : [value]);
     if (initialSearchValues && initialSearchValues.length > 0) {
       const metadata = {};
-      const allOptions = await Promise.all(
-        initialSearchValues.map((v, index) => {
-          return query(forID, collection, [field.get('value_field')], v, file, 1).then(
-            ({ payload }) => {
-              const hits = payload.response?.hits || [];
-              const options = this.parseHitOptions(hits);
-              metadata[v] = hits[0]?.data;
-              return { options, index };
-            },
-          );
-        }),
-      );
+      const searchFieldsArray = getSearchFieldArray(field.get('search_fields'));
+      const { payload } = await query(forID, collection, searchFieldsArray, '', file);
```

Comment:
Thanks @barthc.
Not sending a limit here can cause performance issues when referencing a large collection/file.
However I can't see a way to handle both at the moment.